[Reland][functorch] test for compiling functorch transforms

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #100718

Original PR over at #100151. Was reverted due to internal test failures.
I have fixed the internal build system.

Differential Revision: [D45608453](https://our.internmc.facebook.com/intern/diff/D45608453)

zou3519 committed:
[Reland][functorch] test for compiling functorch transforms

Original PR over at #100151. Was reverted due to internal test failures.
I have fixed the internal build system.

[ghstack-poisoned]

kshitij12345 reviewed (approved):
LGTM, Thanks @zou3519

zou3519 committed:
Update on "[Reland][functorch] test for compiling functorch transforms"


Original PR over at #100151. Was reverted due to internal test failures.
I have fixed the internal build system.

Differential Revision: [D45608453](https://our.internmc.facebook.com/intern/diff/D45608453)

[ghstack-poisoned]

zou3519 commented:
@pytorchbot merge
